src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -96,42 +122,107 @@ void KevinHallModel::update_risk_factors(RuntimeContext &context) {
-        // person.risk_factors[RMR_key] = state.RMR;
-        // person.risk_factors[F_key] = state.F;
-        // person.risk_factors[L_key] = state.L;
-        // person.risk_factors[ECF_key] = state.ECF;
-        // person.risk_factors[G_key] = state.G;
-        // person.risk_factors[W_key] = state.W;
-        // person.risk_factors[EE_key] = state.EE;
-        // person.risk_factors[EI_key] = state.EI;
+        // person.risk_factors["Height"_id] = state.H;
+        // person.risk_factors["Weight"_id] = state.BW;
+        // person.risk_factors["PhysicalActivity"_id] = state.PAL;
+        // person.risk_factors["RestingMetabolicRate"_id] = state.RMR;
+        // person.risk_factors["BodyFat"_id] = state.F;
+        // person.risk_factors["LeanTissue"_id] = state.L;
+        // person.risk_factors["ExtracellularFluid"_id] = state.ECF;
+        // person.risk_factors["Glycogen"_id] = state.G;
+        // person.risk_factors["EnergyExpenditure"_id] = state.EE;
+        // person.risk_factors["EnergyIntake"_id] = state.EI;
+    }
+}
+
+void KevinHallModel::initialise_nutrient_intakes(Person &person) const {
+    auto nutrient_intakes = compute_nutrient_intakes(person);
+
+    // Initialise nutrient intakes.
+    for (auto &[nutrient_key, nutrient_value] : nutrient_intakes) {
+        person.risk_factors[nutrient_key] = nutrient_value;
+    }
+
+    // TODO: set old value to new value (only some nutrients)
+}
+
+void KevinHallModel::update_nutrient_intakes(Person &person) const {
+    auto nutrient_intakes = compute_nutrient_intakes(person);
+
+    // Update nutrient intakes.
+    for (auto &[nutrient_key, nutrient_value] : nutrient_intakes) {
+        person.risk_factors[nutrient_key] = nutrient_value;
+    }
+
+    // TODO: set old value to previous value (only some nutrients)
+}
+
+std::unordered_map<core::Identifier, double>
+KevinHallModel::compute_nutrient_intakes(Person &person) const {
+    std::unordered_map<core::Identifier, double> nutrient_intakes;
+
+    for (const auto &[food_key, nutrient_coefficients] : nutrient_equations_) {
+        double food_intake = person.get_risk_factor_value(food_key);
+        for (const auto &[nutrient_key, nutrient_coefficient] : nutrient_coefficients) {
+            nutrient_intakes[nutrient_key] += food_intake * nutrient_coefficient;
+        }
+    }
+
+    return nutrient_intakes;
+}
+
+void KevinHallModel::initialise_energy_intake(Person &person) const {
+    double energy_intake = compute_energy_intake(person);
+    person.risk_factors["EnergyIntake"_id] = energy_intake;
+
+    // Begin at steady state (EI = EE).
+    person.risk_factors["EnergyExpenditure"_id] = energy_intake;
+
+    // TODO: set old value to new value
+}
+
+void KevinHallModel::update_energy_intake(Person &person) const {
+    double energy_intake = compute_energy_intake(person);
+    person.risk_factors["EnergyIntake"_id] = energy_intake;
+
+    // TODO: set old value to previous value
```

Comment:
Why do you need to set the old value to previous value?

Reply:
This is maybe a bit poorly worded. What I meant to say is save the old value in, say, `"Factor_old"_id` before computing the update of `"Factor"_id`.

Reply:
Why do you need to do that? I mean, what's the old value used for?

Reply:
In Ali's code: `GetFastSteadyStateApproximation`, there's a value `deltaE = EI - EI_old`, or the change in energy intake from last year, which is used to compute body adaptation to changing energy intake rates. I *suppose* it's not strictly necessary to save the old value in `p.risk_factor` as it can be saved locally in the current year's update call before `update_energy_intake` is called, but at the cost of modularity. We have an extra `EI_old` to pass around `update_risk_factors`.

Reply:
What I mean is: we'd have to merge `update_energy_intake` with `simulate_person` in our code.